### PR TITLE
Created genre chart

### PR DESCRIPTION
**Pull request by RazvanBudacaTechquarter**
User can now vote for their favorite genre of music.

After inputting a genre in the form, the server saves the name of the genre and it's number of votes in a map. When fetched, the server sends the map and shows the chart. Might add permanent storage with Datastorage later. Commits might be a bit hectic because I changed my mind halfway, from doing a band chart to a genre chart.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/GenreChartServlet.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    response.setContentType("application/json");
+    String json = gson.toJson(genreVotes);
+    response.getWriter().println(json);
+  }
+
+   @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String genre = request.getParameter("genre");
+    int currentVotes = getCurrentVotes(genreVotes, genre);
+    genreVotes.put(genre, currentVotes + 1);
```
Comment (amaurymedeiros) [inline portfolio/src/main/java/com/google/sps/servlets/GenreChartServlet.java]: You don't need the `getCurrentVotes` method, as you can use the `getOrDefault` API instead... this block of code would then be something like:

`genreVotes.put(genre, genreVotes.getOrDefault(genre, 0) + 1);`

File: portfolio/src/main/webapp/index.html [hunk 2]
```diff
@@ -23,10 +23,12 @@ <h1 id="header">My portfolio</h1>
             <li><a href="projects.html">My projects</a></li>
             <li><a href="bandGen.html">Check out these cool bands</a></li>
           </ul>
+          <p>Enter a comment or 
+              <a href="genreChart.html">vote your favorite genre</a>
```
Comment (amaurymedeiros) [inline portfolio/src/main/webapp/index.html]: nit: vote _for_ your favorite genre

File: portfolio/src/main/java/com/google/sps/servlets/GenreChartServlet.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  private final Gson gson = new Gson();
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("application/json");
+    String json = gson.toJson(genreVotes);
+    response.getWriter().println(json);
+  }
+
+   @Override
```
Comment (jgaillard85) [inline portfolio/src/main/java/com/google/sps/servlets/GenreChartServlet.java]: nit: fix indentation

**Review by jgaillard85 — APPROVED** [review]
You can merge once you've addressed the comments

File: portfolio/src/main/java/com/google/sps/servlets/GenreChartServlet.java (hunk 1 above)
Comment (RazvanBudacaTechquarter) [reply portfolio/src/main/java/com/google/sps/servlets/GenreChartServlet.java]: Totally forgot about that method, thanks!

**Pull request merged**